Modules/Messages/src/Messages.Application/DomainEvents/Incoming/MessageOrphaned/MessageOrphanedDomainEventHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Backbone.Modules.Messages.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Messages.Domain.DomainEvents.Incoming;
+using Backbone.Modules.Messages.Domain.Entities;
+using Microsoft.Extensions.Options;
+
+namespace Backbone.Modules.Messages.Application.DomainEvents.Incoming.MessageOrphaned;
+
+public class MessageOrphanedDomainEventHandler : IDomainEventHandler<MessageOrphanedDomainEvent>
+{
+    private readonly IMessagesRepository _messagesRepository;
+    private readonly ApplicationOptions _applicationOptions;
+
+    public MessageOrphanedDomainEventHandler(IMessagesRepository messagesRepository, IOptions<ApplicationOptions> applicationOptions)
+    {
+        _messagesRepository = messagesRepository;
+        _applicationOptions = applicationOptions.Value;
+    }
+
+    public async Task Handle(MessageOrphanedDomainEvent @event)
+    {
+        await _messagesRepository.Delete(Message.IsMessageOrphaned(_applicationOptions.DidDomainName), CancellationToken.None);
```

Comment:
I agree that I could have given it more thought. The approach of checking all messages was the first solution that came to my mind, and given our previous talk about performance concerns, I thought it would be an ok solution since it would run at night. However, I fully agree that passing the `Id` is a more efficient solution. Changes made.